src/vcs/git.rs

```diff
@@ -1347,22 +1347,81 @@ mod tests {
                 copied: vec![],
                 modified: vec![ModifiedFile {
                     path: Path::with_root(&[unsound::label::new("README.md")]),
-                    diff: FileDiff::Plain(
-                        vec![Hunk {
-                            header: b"@@ -1 +1,2 @@\n".to_vec(),
+                    diff: FileDiff::Plain {
+                        hunks: vec![Hunk {
+                            header: Line(b"@@ -1 +1,2 @@\n".to_vec()),
                             lines: vec![
                                 LineDiff::deletion(b"This repository is a data source for the Upstream front-end tests.\n".to_vec(), 1),
                                 LineDiff::addition(b"This repository is a data source for the Upstream front-end tests and the\n".to_vec(), 1),
                                 LineDiff::addition(b"[`radicle-surf`](https://github.com/radicle-dev/git-platinum) unit tests.\n".to_vec(), 2),
                             ]
                         }]
-                    )
+                    }
                 }]
             };
             assert_eq!(expected_diff, diff);
 
             Ok(())
         }
+
+        #[cfg(feature = "serialize")]
+        #[test]
+        fn test_diff_serde() -> Result<(), Error> {
+            use file_system::*;
+
+            let diff = Diff {
+                created: vec![],
+                deleted: vec![],
+                moved: vec![],
+                copied: vec![],
+                modified: vec![ModifiedFile {
+                    path: Path::with_root(&[unsound::label::new("README.md")]),
+                    diff: FileDiff::Plain {
+                        hunks: vec![Hunk {
+                            header: Line(b"@@ -1 +1,2 @@\n".to_vec()),
+                            lines: vec![
+                                LineDiff::deletion(b"This repository is a data source for the Upstream front-end tests.\n".to_vec(), 1),
+                                LineDiff::addition(b"This repository is a data source for the Upstream front-end tests and the\n".to_vec(), 1),
+                                LineDiff::addition(b"[`radicle-surf`](https://github.com/radicle-dev/git-platinum) unit tests.\n".to_vec(), 2),
+                            ]
+                        }]
+                    }
+                }]
+            };
+
+            let json = serde_json::json!({
+                "created": [],
+                "deleted": [],
+                "moved": [],
+                "copied": [],
+                "modified": [{
```

Comment:
Can we use a diff which has content in these fields for a more thorough assertion?

Reply:
Contributors are welcome :joy:

Reply:
They really are